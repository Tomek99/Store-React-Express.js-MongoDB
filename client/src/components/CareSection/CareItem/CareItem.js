import React from "react";
import PropTypes from "prop-types";
import styles from "./CareItem.module.scss";

function CareItem({ item }) {
  const { url, title, text } = item;
  return (
    <div className={styles.CareItem}>
      <img src={url} alt={title} />
      <div className={styles.wrapperText}>
        <h2>{title}</h2>
        <p>{text}</p>
      </div>
    </div>
  );
}

CareItem.propTypes = {
  url: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
};

export default CareItem;
