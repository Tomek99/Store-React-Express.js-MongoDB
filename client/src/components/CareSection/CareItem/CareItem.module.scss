.CareItem {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 10px;

    img {
        max-width: 210px;
        height: auto;
        width: 100%;
    }

    .wrapperText {
        text-align: center;
        font-size: 1.3rem;
        max-width: 200px;
        line-height: 1.5;

        h2 {
            margin-bottom: 5px;
            font-size: 2rem;
            font-weight: 400;
        }

        p {
            position: relative;
            cursor: pointer;

            &::before {
                content: "";
                position: absolute;
                transition: 0.3s linear;
                left: 0;
                bottom: -3px;
                width: 0;
                height: 2px;
                background-color: #000;
            }

            &:hover::before {
                width: 100%;
            }
        }
    }
}
