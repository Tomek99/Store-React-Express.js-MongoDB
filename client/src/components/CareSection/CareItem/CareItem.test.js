import { render, screen } from "@testing-library/react";
import CareItem from "./CareItem";

test("Shoulde render CareItem component", () => {
  const item = {
    url: "https://res.cloudinary.com/dvoduabha/image/upload/v1679933916/we_care/protect_ofonsc.webp",
    title: "title_test",
    text: "text_title",
  };

  const { container } = render(<CareItem item={item} />);

  expect(container).toMatchSnapshot();
});
