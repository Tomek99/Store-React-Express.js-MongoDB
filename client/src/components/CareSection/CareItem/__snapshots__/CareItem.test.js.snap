// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoulde render CareItem component 1`] = `
<div>
  <div
    class="CareItem"
  >
    <img
      alt="title_test"
      src="https://res.cloudinary.com/dvoduabha/image/upload/v1679933916/we_care/protect_ofonsc.webp"
    />
    <div
      class="wrapperText"
    >
      <h2>
        title_test
      </h2>
      <p>
        text_title
      </p>
    </div>
  </div>
</div>
`;
